mule-extensions-api/src/main/java/org/mule/runtime/extension/internal/loader/enricher/ConfigRefDeclarationEnricher.java

```diff
@@ -102,15 +105,16 @@ protected void onConfiguration(ConfigurationDeclaration config) {
     return componentConfigs;
   }
 
-  private ParameterDeclaration buildConfigRefParameter(Collection<ConfigurationDeclaration> configs) {
+  private ParameterDeclaration buildConfigRefParameter(ComponentDeclaration componentDeclaration,
+                                                       Collection<ConfigurationDeclaration> configs) {
     ParameterDeclaration parameter = new ParameterDeclaration(CONFIG_REF_NAME);
     parameter.setDescription("The name of the configuration to be used to execute this component");
     parameter.setRequired(!hasAnImplicitConfig(configs));
     parameter.setParameterRole(BEHAVIOUR);
     parameter.addModelProperty(new SyntheticModelModelProperty());
     parameter.setDslConfiguration(ParameterDslConfiguration.builder().allowsReferences(true).build());
     parameter.setType(CONFIG_TYPE, false);
-    parameter.setExpressionSupport(NOT_SUPPORTED);
+    parameter.setExpressionSupport(expressionSupport(componentDeclaration));
```

Comment:
this should be Java and Mule SDK specific. Do not add additional behavior to XML SDK

Reply:
This would mean that xml-sdk connectors(like existing rest-connect connectors) will not be able to be used within the Mule SDK operations unless the configs used on their operations are statically referenced. Is this ok?

Reply:
ok, good point.